flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultScheduler.java

```diff
@@ -246,23 +256,23 @@ public void allocateSlotsAndDeploy(final Collection<ExecutionVertexDeploymentOpt
 		}
 	}
 
+	private void validateDeploymentOptions(final Collection<ExecutionVertexDeploymentOption> deploymentOptions) {
+		deploymentOptions.stream()
+			.map(ExecutionVertexDeploymentOption::getExecutionVertexId)
+			.map(this::getExecutionVertex)
+			.map(ExecutionVertex::getExecutionState)
+			.forEach(state -> checkState(
+				state == ExecutionState.CREATED,
+				"vertex to schedule must be in CREATED state"));
```

Comment:
I think it's better to also include the execution vertex id in the error message, e.g., _"expected execution of vertex %s to be in CREATED state, was: %s"_

Reply:
Ok. Done.